lib/DependencyScan/ModuleDependencyCacheSerialization.cpp

```diff
@@ -1202,6 +1241,16 @@ void ModuleDependenciesCacheSerializer::writeMetadata(StringRef scanningContextH
                              scanningContextHash);
 }
 
+void ModuleDependenciesCacheSerializer::writeSerializationTime() {
+  using namespace graph_block;
+  llvm::sys::TimePoint<> now = std::chrono::system_clock::now();
```

Comment:
We only have one timestamp to guard the entire graph? I think there is a time window here between the input is read and graph is serialized, that if the input is modified, the scan will mistakenly be treated as upToDate.

Reply:
That's a good catch, thank you. I moved the serialization time-stamp to be generated in the constructor of `ModuleDependenciesCache` which we create at the beginning of each query.